### PR TITLE
Implement exclusions in rsync file copier

### PR DESCRIPTION
**Pull request by TravisCarden**

### CODE REVIEW

File: config/services.yml [hunk 1]
```diff
@@ -8,14 +8,10 @@ services:
 
     PhpTuf\ComposerStager\:
         resource: '../src/*'
-        exclude:
-            - '../src/Exception'
-
-    PhpTuf\ComposerStager\Console\:
-        resource: '../src/Console/*'
         public: true
```
Comment (TravisCarden) [inline config/services.yml:11]: Alas, I had to make all classes public in order to access a specific file copier in functional tests. 😕

File: tests/Functional/RsyncFileCopierTest.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            'sit-exclude',
+        ]);
+
+        self::assertFileExists(self::ACTIVE_DIR . '/lorem.txt');
+        self::assertFileExists(self::ACTIVE_DIR . '/ipsum.txt');
+        self::assertFileDoesNotExist(self::STAGING_DIR . '/dolor-exclude.txt');
+        self::assertFileDoesNotExist(self::STAGING_DIR . '/sit-exclude');
+    }
+
+    /**
```
Comment (TravisCarden) [inline tests/Functional/RsyncFileCopierTest.php:77]: This test isn't exactly in scope. I added it for consistency with the Symfony file copier functional tests. ¯\_(ツ)_/¯

File: tests/Functional/RsyncFileCopierTest.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        if (!self::isRsyncAvailable()) {
+            $this->markTestSkipped('Rsync is not available for testing.');
+        }
+    }
+
+    protected static function isRsyncAvailable(): bool
+    {
+        $finder = new ExecutableFinder();
+        return !($finder->find('rsync') === null);
```
Comment (phenaproxima) [inline tests/Functional/RsyncFileCopierTest.php:46]: Wouldn't something like this be a little clearer?
```suggestion
        return $finder->find('rsync') !== null;
```

File: tests/Functional/RsyncFileCopierTest.php [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        $sut = $container->get(RsyncFileCopier::class);
+        return $sut;
+    }
+
+    public function testCopy(): void
+    {
+        touch(self::ACTIVE_DIR . '/lorem.txt');
+        touch(self::ACTIVE_DIR . '/ipsum.txt');
+        touch(self::ACTIVE_DIR . '/dolor-exclude.txt');
+        mkdir(self::ACTIVE_DIR . '/sit-exclude');
```
Comment (phenaproxima) [inline tests/Functional/RsyncFileCopierTest.php:63]: `touch()` and `mkdir()` return TRUE on success and FALSE on failure, so would it maybe make sense to wrap these calls in `self::assertTrue()`?

**Review by phenaproxima — APPROVED** [review]
Approved, just a couple nits.

File: tests/Functional/RsyncFileCopierTest.php (hunk 3 above)
Comment (TravisCarden) [reply tests/Functional/RsyncFileCopierTest.php:46]: Yes, I think so.

File: tests/Functional/RsyncFileCopierTest.php (hunk 4 above)
Comment (TravisCarden) [reply tests/Functional/RsyncFileCopierTest.php]: Sure. If this fail so would the rest of the test, but some assertions here would probably help with defect localization in that case.

**Pull request merged**